appium/lib/screen-object.rb

```diff
@@ -66,84 +69,142 @@ def wait_step(timeout = 5, message = nil, &block)
-        false
-      end
-    end
+    driver.send_keys(:enter)
   end
 
-  def scroll_down_click(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          driver.find_element(locator,locator_value).click
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollStart,:end_x =>0,:end_y =>scrollEnd,:touchCount => 2,:duration => 0)
-        false
-      end
-    end
+  def gesture(arg)
+    Appium::TouchAction.new(driver).swipe(start_x: arg[0], start_y: arg[1], end_x: arg[2], end_y: arg[3], duration: arg[4]).perform
+  rescue RuntimeError => e
+    raise("Error during gesture \n Error Details: #{e}")
   end
 
-  def scroll_up_find(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollEnd,:end_x =>0,:end_y =>scrollStart,:touchCount => 2,:duration => 0)
-        false
-      end
-    end
+  # Scrolls device screen in a direction
+  #
+  # @param direction [symbol] The direction to search for an string
+  # @param duration   [Integer] The amount of times in seconds we want to scroll to find the element
+  def scroll(direction = :down, duration = 1000)
+    size = driver.window_size
+    x = size.width / 2
+    y = size.height / 2
+    loc = case direction
+          when :up then    [x, y * 0.5, x, (y + (y * 0.3)), duration]
+          when :down then  [x, y, x, y * 0.5, duration]
+          when :left then  [x * 0.6, y, x * 0.3, y, duration]
+          when :right then [x * 0.3, y, x * 0.6, y, duration]
+          else
+            raise('Only up, down, left and right scrolling are supported')
+          end
+    gesture(loc)
+  end
+
+  def scroll_down
+    scroll(:down)
   end
 
+  def scroll_up
+    scroll(:up)
+  end
+
+  def swipe_left
+    scroll(:left)
+  end
+
+  def swipe_right
+    scroll(:right)
+  end
+
+  # Swipe a screen element Down, Up, Left and Right direction
+  #
+  # @param locator      [Screen-object] The screen element you are looking for on the screen
+  # @param direction    [symbol] By default is down. up, left amd right are also available
+  # @param duration     [integer] The timeout value for element you are looking for on the screen
+  def swipe_screen_element(locator, direction = :down, duration = 1000)
```

Comment:
Remove this method from here and keep the one in `ScreenObject::AppElements::Element`